app/src/main/cpp/skyline/vfs/nacp.cpp

```diff
@@ -7,15 +7,26 @@ namespace skyline::vfs {
     NACP::NACP(const std::shared_ptr<vfs::Backing> &backing) {
         nacpContents = backing->Read<NacpData>();
 
-        // TODO: Select based on language settings, complete struct, yada yada
-
-        // Iterate till we get to the first populated entry
+        u32 counter{0};
         for (auto &entry : nacpContents.titleEntries) {
-            if (entry.applicationName.front() == '\0')
-                continue;
-
-            applicationName = span(entry.applicationName).as_string(true);
-            applicationPublisher = span(entry.applicationPublisher).as_string(true);
+            if (entry.applicationName.front() != '\0') {
```

Comment:
I think there's an explicit field for this? Maybe I'm wrong though

Reply:
Also an index based for loop would be better here if there isn't a specific field for this

Reply:
explicit field? I'm not sure what you're referring to

Reply:
in the nacp struct, theres a bitmask